MaeBlock.cpp

```diff
@@ -333,42 +335,34 @@ bool IndexedProperty<double>::operator==(const IndexedProperty<double>& rhs) con
 {
     if(m_is_null == nullptr || rhs.m_is_null == nullptr) {
         if((m_is_null == nullptr) != (rhs.m_is_null == nullptr)) return false;
-    } else if(*m_is_null != *(rhs.m_is_null)) {
-        return false;
-    }
-    for(int i=0; i<m_data.size(); i++)
+    } else if(*m_is_null != *(rhs.m_is_null)) return false;
+
+    for(int i=0; i<m_data.size(); ++i)
         if(abs(m_data[i] - rhs.m_data[i]) > tolerance) return false;
 
     return true;
 }
 
-bool IndexedBlock::operator==(const IndexedBlock& rhs) const
+template <typename T>
+static bool maps_indexed_props_equal(const T& lmap, const T& rmap)
 {
-    using namespace std;
-
-    for(const auto& p : m_bmap) {
-        if(rhs.m_bmap.count(p.first) != 1) return false;
-        const auto& bmap2 = rhs.m_bmap.find(p.first);
-        if(!(*(bmap2->second) == *(p.second))) return false;
+    for(const auto& p : lmap) {
```

Comment:
Why does this loop over lmap? As far as I can tell, `p` isn't used in the search.

Reply:
Because I wrote something terrible!  thanks for catching